### DISCUSSION TITLE
What is the best way to fetch the data if not authenticated (check if user email has been verified)

### DISCUSSION THREAD

**Discussion by ngamsomset** [supabase/supabase#29194] [Questions] [answered]
Hello there

I'm seeking suggestions on how to fetch the data from the table that has RLS enabled. 

Situation
I'm trying to check for one of the column in the `users_profile`  table where there is one column `email_confirmed_at`. This is the check only to `verified` user can login. You normally can't fetch the data from any table with RLS enabled if you are not authenticated which as a non-login user, you are not authenticated.

Outcome
I'd want to be able to check if the user has been verified their account before they can login.

What is the best approach here? 

I have a few that came across my mind. 
1. Create View just only for that column. (look like a workaround RLS, maybe not sustainable?)
2. Fetch the data as admin user, possible? (service role key is not recommended on the client side)
3. with that RLS but create a policy to allow anyone to SELECT. (ideal? not safe? not sure the different between disabled RLS and allow anyone to access with a policy)

Would appreciate any suggestion, thank you!

**Comment by GaryAustin1** [comment, marked as answer]
You can also use an rpc call to a Postgres security definer function to bypass RLS and return info. 
The problem with any of these methods if they can be called from a browser with anon user is anyone can call them so you really need to think thru if that is what you want.  Otherwise you would need to run server side as service_role.

**Reply by ngamsomset** [reply]
thanks for your answer. Just want to sum it up. All of the mentioned method including RPC call with security definer also not a "safe" way to make a call to the database as it bypass RLS. The best way to do this is to do a server side call with service_role.
